src/mp4box/mp4a.rs

```diff
@@ -564,11 +570,18 @@ impl<R: Read + Seek> ReadDesc<&mut R> for DecoderSpecificDescriptor {
 
 impl<W: Write> WriteDesc<&mut W> for DecoderSpecificDescriptor {
     fn write_desc(&self, writer: &mut W) -> Result<u32> {
-        let size = Self::desc_size();
+        let size = self.desc_size();
         write_desc(writer, Self::desc_tag(), size)?;
 
-        writer.write_u8((self.profile << 3) + (self.freq_index >> 1))?;
-        writer.write_u8((self.freq_index << 7) + (self.chan_conf << 3))?;
+        if self.profile < 31 {
```

Comment:
A potential option could be to use some crate like:
https://docs.rs/bitstream-io/1.7.0/bitstream_io/
for this bitpacking